tests/glsl-intrinsic/atomic/atomicStorageBuffer.slang

```diff
@@ -0,0 +1,318 @@
+        && atomicCompSwap(uint64Buffer.data[0], 5, 2) == 5
+        && uint64Buffer.data[0] == 2
+
+        && (uint64Buffer.data[0] = 5) == 5
+        && atomicCompSwap(uint64Buffer.data[0], 4, 2) == 5
+        && uint64Buffer.data[0] == 5
+        ;
+}
+
+bool testAtomicFloat16()
+{
+    return true
+#ifdef TEST_when_shader_atomic_float2_is_available
+        && (float16Buffer.data[0] = half(5)) == half(5)
+        && atomicAdd(float16Buffer.data[0], half(1)) == half(5)
+        && float16Buffer.data[0] == half(6)
+
+        && (float16Buffer.data[0] = half(5)) == half(5)
+        && atomicMin(float16Buffer.data[0], half(1)) == half(5)
+        && float16Buffer.data[0] == half(1)
+
+        && (float16Buffer.data[0] = half(5)) == half(5)
+        && atomicMax(float16Buffer.data[0], half(1)) == half(5)
+        && float16Buffer.data[0] == half(5)
+
+        && (float16Buffer.data[0] = half(5)) == half(5)
+        && atomicExchange(float16Buffer.data[0], half(2)) == half(5)
+        && float16Buffer.data[0] == half(2)
+#endif // TEST_when_shader_atomic_float2_is_available
+        ;
+}
+
+bool testAtomicFloat32()
+{
+    return true
+
+        && (float32Buffer.data[0] = float(5)) == float(5)
+        && atomicAdd(float32Buffer.data[0], float(1)) == float(5)
+        && float32Buffer.data[0] == float(6)
+
+#ifdef TEST_when_shader_atomic_float2_is_available
+        && (float32Buffer.data[0] = float(5)) == float(5)
+        && atomicMin(float32Buffer.data[0], float(1)) == float(5)
+        && float32Buffer.data[0] == float(1)
+
+        && (float32Buffer.data[0] = float(5)) == float(5)
+        && atomicMax(float32Buffer.data[0], float(1)) == float(5)
+        && float32Buffer.data[0] == float(5)
+
+        && (float32Buffer.data[0] = float(5)) == float(5)
+        && atomicExchange(float32Buffer.data[0], float(2)) == float(5)
+        && float32Buffer.data[0] == float(2)
+#endif // TEST_when_shader_atomic_float2_is_available
+        ;
+}
+
+bool testAtomicFloat64()
+{
+    return true
+
+        && (float64Buffer.data[0] = double(5)) == double(5)
+        && atomicAdd(float64Buffer.data[0], double(1)) == double(5)
+        && float64Buffer.data[0] == double(6)
+
+#ifdef TEST_when_shader_atomic_float2_is_available
+        && (float64Buffer.data[0] = double(5)) == double(5)
+        && atomicMin(float64Buffer.data[0], double(1)) == double(5)
+        && float64Buffer.data[0] == double(1)
+
+        && (float64Buffer.data[0] = double(5)) == double(5)
+        && atomicMax(float64Buffer.data[0], double(1)) == double(5)
+        && float64Buffer.data[0] == double(5)
+
+        && (float64Buffer.data[0] = double(5)) == double(5)
+        && atomicExchange(float64Buffer.data[0], double(2)) == double(5)
+        && float64Buffer.data[0] == double(2)
+#endif // TEST_when_shader_atomic_float2_is_available
+        ;
+}
+
+layout(local_size_x = 1) in;
+void computeMain()
+{
+    // testing has the following pattern in groups:
+    // set, op, test
```

Comment:
I am not sure what this means.
Did you mean that "testAtomicInt32()", as an example, runs tests in a certain order?
I am not sure why it is important to be a comment.
And "set" sounds like a mutating functions, and "test" sounds like just testing the values but I cannot guess what "op" means.

Reply:
I will reword the comment